packages/multicast_dns/test/client_test.dart

```diff
@@ -10,11 +10,11 @@ import 'package:test/test.dart';
 
 void main() {
   test('Can inject datagram socket factory and configure mdns port', () async {
-    int lastPort;
+    int? lastPort;
     final MockRawDatagramSocket mockRawDatagramSocket = MockRawDatagramSocket();
```

Comment:
`mockRawDatagramSocket.address` is Null somehow, not sure why. 

test fails with `Null is not a subtype of type  InternetAddress`

Reply:
Mockito doesn't mix very well with nullability; see https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md

How hard would this class be to mock manually?

Reply:
Let me try

Reply:
Trying to wrap my head around this.

Reply:
will try to implement the manual way.